packages/pluggableWidgets/notifications-native/src/Notifications.tsx

```diff
@@ -1,67 +1,128 @@
-import { ActionValue, ValueStatus } from "mendix";
-import { useEffect, useRef, useState } from "react";
 import messaging, { FirebaseMessagingTypes } from "@react-native-firebase/messaging";
+import PushNotification from "react-native-push-notification";
+import { executeAction } from "@mendix/piw-utils-internal";
+import { useCallback, useEffect, useRef, useState } from "react";
+import { ActionValue, ValueStatus } from "mendix";
 import "@react-native-firebase/app";
 
 import { ActionsType, NotificationsProps } from "../typings/NotificationsProps";
-import { executeAction } from "@mendix/piw-utils-internal";
+
+declare type Option<T> = T | undefined;
+
+// re-declare the library's type because: 1) it doesn't match library version 2) the definition file exports two symbols with same name.
+interface IPushNotification {
+    playSound?: boolean;
+    subText?: string;
+    userInteraction: boolean;
+    id?: number;
+    title: string;
+    message: string;
+    channelId?: string;
+    foreground: boolean;
+    data: NotificationData & {
+        actionIdentifier?: string;
+        userInteraction?: number;
+    }; // iOS
+}
 
 interface NotificationData {
     actionName?: string;
     guid?: string;
 }
 
-export function Notifications(props: NotificationsProps<undefined>) {
+interface Helpers {
```

Comment:
Wouldn't it be simpler to define a structure with the appropriate properties, and then fill those properties using the right strategy (i.e. taking them from the `notification` object for remote notifications, and calling the helper functions for local notifications)?

Reply:
Indeed; makes thing simpler. No need for the function and also squashed two arguments into one object.

Reply:
@KevinVlaanderen updated

Reply:
Wait, I think I fully get what you mean now. My commit improved things but I think I can get it a bit better. One sec

Reply:
Nevermind, it's good :)